File: server/View.Server.Shared/Data/Room.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace View.Server.Shared.Data
{
    public class Room
    {
        public int Id { get; set; }
        
        public string Name { get; set; }
        
        public int MaxPersonCount { get; set; }
        
        public int PosX { get; set; }
        public int PosY { get; set; }
        
        public int Width { get; set; }
        public int Height { get; set; }
        
        public string IoTDeviceId { get; set; }
        
        [JsonIgnore]
        public IoTDevice IoTDevice { get; set; }
        
        public int FloorId { get; set; }
        
        [JsonIgnore]
        public Floor Floor { get; set; }

        [JsonIgnore]
        public List<Occupation> Occupations { get; set; }
        
    }
}

File: frontend/View/src/app/models/room.ts

export class Room {

  constructor(
    public name: string,
    public maxPersonCount: number,
    public floorId: number,

    public posX: number,
    public posY: number,
    public width: number,
    public height: number,

    public id?: number,
    public iotDeviceId?: number,

  ) { }
}

export function isRoom(arg: any): arg is Room {
  if (typeof arg.id !== 'number') { return false; }
  if (typeof arg.name !== 'string') { return false; }
  if (typeof arg.maxPersonCount !== 'number') { return false; }
  if (typeof arg.floorId !== 'number') { return false; }

  if (typeof arg.posX !== 'number') { return false; }
  if (typeof arg.posY !== 'number') { return false; }
  if (typeof arg.width !== 'number') { return false; }
  if (typeof arg.height !== 'number') { return false; }

  return true;
}

export function isRoomArray(arg: any): arg is Room[] {
  if (!Array.isArray(arg)) { return false; }
  for (const floor of arg) {
    if (!isRoom(floor)) { return false; }
  }
  return true;
}

